zio-http/src/main/scala/zhttp/socket/Socket.scala

```diff
@@ -20,6 +20,9 @@ sealed trait Socket[-R, +E, -A, +B] { self =>
 
   private[zhttp] def execute(a: A): ZStream[R, E, B] = self(a)
 
+  def provide(r: R)(implicit needsEnv: NeedsEnv[R]): Socket[Any, E, A, B] =
+    Socket.FromStreamingFunction(a => self(a).provide(r))
```

Comment:
```suggestion
    Socket.Provide(r, self)
```

Reply:
While providing an implementation for the Provide type and the method, the compiler keeps on warning that the effect has **Any** as its requirement
```
def apply(a: A): ZStream[R, E, B] = self match {
...
    case Provide(m, r)               => m(a).provide(r)
}
def provide(r: R)(implicit needsEnv: NeedsEnv[R]): Socket[R, E, A, B] = Provide(self, r)
private final case class Provide[R, E, A, B](m: Socket[R, E, A, B], r: R)                 extends Socket[R, E, A, B]
```

I believe the intention behind your comment is that with provide we should return a declaration of what we'd end up with and then the apply method is where the actual environment elimination should be.

Reply:
This has been resolved in newer commit(s)